ipaserver/install/server/replicainstall.py

```diff
@@ -904,7 +904,11 @@ def promote_check(installer):
                            "certificate")
 
     installutils.verify_fqdn(config.host_name, options.no_host_dns)
-    installutils.verify_fqdn(config.master_host_name, options.no_host_dns)
+    # Inside the container environment master's IP address does not
+    # resolve to its name. See bz #1364139
+    container_environment = tasks.detect_container()
+    installutils.verify_fqdn(config.master_host_name, options.no_host_dns,
+                             local_hostname=not container_environment)
```

Comment:
I'm not sure the test is right. detect_container() returns None if not a container but a string value if it does (e.g. container-other). So perhaps:

container_environment = tasks.detect_container() is not None

Reply:
Oh yes this or `bool(tasks.detect_container())` You decide.